hack/e2e.sh

```diff
@@ -13,31 +13,36 @@ fi
+  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[EndUser\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e "${ARTIFACT_FLAG:-}"
   oc logout
 fi
 
 if [[ -z "$SUITE" || "$SUITE" == "clusterreader" ]]; then
   echo "Running azure cluster reader e2e tests"
   (awk '/^  azureClusterReaderKubeconfig:/ { print $2 }' <_data/containerservice.yaml | base64 -d) > _data/_out/azure-cluster-reader.kubeconfig
-  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[AzureClusterReader\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e -kubeconfig ../../_data/_out/azure-cluster-reader.kubeconfig
+  export KUBECONFIG=../../_data/_out/azure-cluster-reader.kubeconfig
+  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[AzureClusterReader\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e
 fi
 
 if [[ -z "$SUITE" || "$SUITE" == "customer-cluster-admin" ]]; then
   echo "Running azure customer-cluster-admin tests"
   fqdn=$(awk '/^  fqdn:/ { print $2 }' <_data/containerservice.yaml)
-  export KUBECONFIG_CUSTOMER_ADMIN=_data/_out/customer-cluster-admin.kubeconfig
-  export KUBECONFIG_CUSTOMER_READER=_data/_out/customer-cluster-reader.kubeconfig
-  export KUBECONFIG_END_USER=_data/_out/enduser.kubeconfig
-
-  # oc login is going to create the kubeconfig with the proper permissions
-  KUBECONFIG=$KUBECONFIG_CUSTOMER_ADMIN oc login $fqdn --username customer-cluster-admin --password "$(awk '/^  customerAdminPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
-  KUBECONFIG=$KUBECONFIG_CUSTOMER_READER oc login $fqdn --username customer-cluster-reader --password "$(awk '/^  customerReaderPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
-  KUBECONFIG=$KUBECONFIG_END_USER oc login $fqdn --username enduser --password "$(awk '/^  endUserPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
-  go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[CustomerAdmin\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e -kubeconfig ../../_data/_out/enduser.kubeconfig "${ARTIFACT_FLAG:-}"
+
+  export KUBECONFIG=_data/_out/customer-cluster-admin.kubeconfig
+  oc login $fqdn --username customer-cluster-admin --password "$(awk '/^  customerAdminPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
+  oc config rename-context $(oc config current-context) customer-cluster-admin
+  oc login $fqdn --username customer-cluster-reader --password "$(awk '/^  customerReaderPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
+  oc config rename-context $(oc config current-context) customer-cluster-reader
+  oc login $fqdn --username enduser --password "$(awk '/^  endUserPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
+  oc config rename-context $(oc config current-context) enduser
+
+  export KUBECONFIG=../../_data/_out/customer-cluster-admin.kubeconfig
```

Comment:
L38 and L46 is little bit confusing. Im ok with this, but we should think how to simplify this and not to have these walking up and down the directories :/